src/acceleration/AitkenAcceleration.cpp

```diff
@@ -80,7 +80,6 @@ void AitkenAcceleration::performAcceleration(
     // preconditioner not necessary
     _aitkenFactor = math::sign(_aitkenFactor) * std::min(_initialRelaxation, std::abs(_aitkenFactor));
   } else {
-    // Todo: make this conditional for dataIDs > 1
```

Comment:
@uekerman this leftover TODO doesn't make sense to me, what do you think?

Also, we apply the scaling ('preconditioning') to data used for the computation of the `_aitkenFactor`) and not for the coupling data itself. Is that correct? I think we discussed that in the past, but I'm asking once more to be sure.

Reply:
You only need the "preconditioner" if you have more than one data field.
But does neither hurt to apply it.

> Also, we apply the scaling ('preconditioning') to data used for the computation of the _aitkenFactor) and not for the coupling data itself. Is that correct?

Exactly

Reply:
I made this now conditional and (please check) applied the scaling also in the first iteration, i.e., put the preconditioner out of the if statement. I think that's required for residual(sum) preconditioners which incorporate iteration history data.

Reply:
In the first iteration `residualDeltas` and `_oldResiduals` are not used, so I don't think we need it here.
But shouldn't hurt either.

Reply:
It's not used in the `AitkenAcceleration` (is it what you mean?), but in the preconditioner class, e.g.,

Reply:
OK, I think I now understand what you mean. The `update` should indeed happen in every iteration. The `apply` only when needed.

Reply:
Yes, I thought it's not a big deal, moved it now back.